cmd/claimserver/endpoint/claim.go

```diff
@@ -1,66 +1,42 @@
 package endpoint
 
 import (
-	"errors"
 
+	"github.com/ethereum/go-ethereum/common"
 	"github.com/gin-gonic/gin"
 	"github.com/iden3/go-iden3/cmd/genericserver"
 	common3 "github.com/iden3/go-iden3/common"
 	"github.com/iden3/go-iden3/core"
+	"github.com/iden3/go-iden3/utils"
 
 	"github.com/iden3/go-iden3/merkletree"
 )
 
 type claimData struct {
-	IndexData string
-	Data      string
+	IdAddr  	common.Address `json:"idData" binding:"required"`
+	Data      string         `json:"data" binding:"required"`
 }
 
 // handlePostClaim handles the request to add a claim to a user tree.
 func handlePostClaim(c *gin.Context) {
 	var m claimData
-	c.BindJSON(&m)
-
-	indexData, err := common3.HexDecode(m.IndexData)
-	if err != nil {
-		genericserver.Fail(c, "error on handlePostClaim", err)
-		return
-	}
-	data, err := common3.HexDecode(m.Data)
-	if err != nil {
-		genericserver.Fail(c, "error on handlePostClaim", err)
-		return
-	}
-
-	if len(indexData) < 400/8 {
-		genericserver.Fail(c, "error on handlePostClaim", errors.New("indexData smaller than 400/8"))
-		return
-	}
-	if len(data) < 496/8 {
-		genericserver.Fail(c, "error on handlePostClaim", errors.New("data smaller than 496/8"))
+	if err := c.BindJSON(&m); err != nil {
+		genericserver.Fail(c, "cannot parse json body", err)
 		return
 	}
 
-	var indexSlot [400 / 8]byte
-	var dataSlot [496 / 8]byte
-	copy(indexSlot[:], indexData[:400/8])
-	copy(dataSlot[:], data[:496/8])
-	claim := core.NewClaimBasic(indexSlot, dataSlot)
-	err = genericserver.Claimservice.AddDirectClaim(*claim)
-	if err != nil {
-		genericserver.Fail(c, "error on AddDirectClaim", err)
-		return
-	}
+	data := []byte(m.Data)
+	hash := utils.HashBytes([]byte(data))
+	claim := core.NewClaimLinkObjectIdentity(0, 6, 0, m.IdAddr, merkletree.Hash(hash))
```

Comment:
What are all these numbers? 0, 6, 0

Reply:
add `ObjectType` and `HashType` as uint32 to define constants